int32[] ports
string[] json_settings
bool render
string address